src/hotspot/share/opto/predicates.hpp

```diff
@@ -963,7 +963,7 @@ class NodeInOriginalLoopBody : public NodeInLoopBody {
 
   // Check if 'node' is not a cloned node (i.e. "< _first_node_index_in_cloned_loop_body") and if we've created a
   // clone from 'node' (i.e. _old_new entry is non-null). Then we know that 'node' belongs to the original loop body.
-  bool check(Node* node) const override {
+  bool check_node_in_loop_body(Node* node) const override {
     if (node->_idx < _first_node_index_in_cloned_loop_body) {
       Node* cloned_node = _old_new[node->_idx];
       return cloned_node != nullptr && cloned_node->_idx >= _first_node_index_in_cloned_loop_body;
```

Comment:
In what case would we return false here? Can you add a comment?

Reply:
Added!